x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/elser/HuggingFaceElserServiceSettings.java

```diff
@@ -26,42 +29,63 @@
 import static org.elasticsearch.xpack.inference.services.ServiceUtils.createUri;
 import static org.elasticsearch.xpack.inference.services.huggingface.HuggingFaceServiceSettings.extractUri;
 
-public record HuggingFaceElserServiceSettings(URI uri, Integer maxInputTokens)
```

Comment:
It's hard to do `if-else` blocks in the constructors if this is a record. I ended up taking a different approach but I left the changes to convert this to a regular class anyway.